subprojects/resources/src/main/java/org/gradle/internal/resource/MissingFileTextResource.java

```diff
@@ -20,15 +20,19 @@
 import org.gradle.internal.hash.HashCode;
 import org.gradle.internal.hash.Hashing;
 
+import javax.annotation.Nonnull;
 import java.io.File;
 import java.io.Reader;
 import java.io.StringReader;
 import java.nio.charset.Charset;
 
-public class EmptyFileTextResource extends UriTextResource {
-    private static final HashCode SIGNATURE = Hashing.signature(EmptyFileTextResource.class);
+/**
+ * A {@link UriTextResource} for a file location but that does not actually exist in the file system.
+ */
+public class MissingFileTextResource extends UriTextResource {
+    private static final HashCode SIGNATURE = Hashing.signature(MissingFileTextResource.class);
```

Comment:
Renamed to more closely reflect the use case for this class - to model a non-null file location that is known not to exist in the file system